redis/src/cluster_async/connections_logic.rs

```diff
@@ -53,48 +54,406 @@ pub(crate) async fn get_or_create_conn<C>(
+                    node: create_async_node(user_conn, None, user_ip),
+                    err,
+                },
+                Err(err) => err.into(),
+            }
+        }
+        (Err(err_1), Err(err_2)) => {
+            // Neither of the connections succeeded.
+            RedisError::from((
+                ErrorKind::IoError,
+                "Failed to refresh both connections",
+                format!(
+                    "Node: {:?} received errors: `{:?}`, `{:?}`",
+                    addr, err_1, err_2
+                ),
+            ))
+            .into()
+        }
+    }
+}
+
+async fn create_management_connection<C>(
+    addr: &str,
+    params: ClusterParams,
+    socket_addr: Option<SocketAddr>,
+) -> RedisResult<(C, Option<IpAddr>)>
+where
+    C: ConnectionLike + Connect + Send + Sync + 'static + Clone,
+{
+    let res = create_connection(addr, params.clone(), socket_addr).await;
+    if let Err(err) = res.as_ref() {
+        warn_management_conn_faild(addr, err);
+    };
+    res
+}
+
+async fn connect_and_check_only_management_conn<C>(
+    addr: &str,
+    params: ClusterParams,
+    socket_addr: Option<SocketAddr>,
+    prev_node: AsyncClusterNode<C>,
+) -> ConnectAndCheckResult<C>
+where
+    C: ConnectionLike + Connect + Send + Sync + 'static + Clone,
+{
+    let (mut new_conn, new_ip) =
+        match create_management_connection(addr, params.clone(), socket_addr).await {
+            Ok(tuple) => tuple,
+            Err(err) => {
+                let node = ClusterNode {
+                    management_connection: None,
+                    ..prev_node
+                };
+                return ConnectAndCheckResult::ManagementConnectionFailed { node, err };
+            }
+        };
+
+    let (user_connection, mut management_connection) = if new_ip != prev_node.ip {
+        // An IP mismatch was detected. Attempt to establish a new connection to replace both the management and user connections.
+        // Use the successfully established connection for the user, then proceed to create a new one for management.
+        warn_mismatch_ip(addr, new_ip, prev_node.ip);
+        if let Err(err) = setup_user_connection(&mut new_conn, params.clone()).await {
+            return ConnectAndCheckResult::Failed(err);
+        }
+        let user_connection = async { new_conn }.boxed().shared();
+        let management_connection = create_management_connection(addr, params.clone(), socket_addr)
+            .await
+            .map(|(conn, _ip)| conn)
+            .ok();
+        (user_connection, management_connection)
+    } else {
+        (prev_node.user_connection, Some(new_conn))
+    };
+
+    if let Some(new_conn) = management_connection.as_mut() {
+        // The new IP matches the existing one. Use this connection for the management connection.
+        // TODO - should we fail here even if a new user connection was set up? or in that situation, should we return the new connection?
```

Comment:
Why should we fail? if we have a working user connection, I think ManagementConnectionFailed is exactly what needed. Maybe we can increase the retries for setting up the management/user connection, as it is done after a connection is already esatblished